internal/commands/kubernetes/nodegroup/create.go

```diff
@@ -0,0 +1,179 @@
+package nodegroup
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/completion"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/output"
+	"github.com/UpCloudLtd/upcloud-cli/v2/internal/resolver"
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud/request"
+	"github.com/spf13/pflag"
+)
+
+type CreateNodeGroupParams struct {
+	Count       int
+	Name        string
+	Plan        string
+	SSHKeys     []string
+	Storage     string
+	KubeletArgs []string
+	Labels      []string
+	Taints      []string
+}
+
+func GetCreateNodeGroupFlagSet(p *CreateNodeGroupParams) *pflag.FlagSet {
+	fs := &pflag.FlagSet{}
+
+	fs.IntVar(&p.Count, "count", 0, "")
+	fs.StringArrayVar(&p.KubeletArgs, "kubelet-arg", []string{}, "")
+	fs.StringArrayVar(&p.Labels, "label", []string{}, "")
+	fs.StringVar(&p.Name, "name", "", "")
+	fs.StringVar(&p.Plan, "plan", "", "")
+	fs.StringArrayVar(&p.SSHKeys, "ssh-key", []string{}, "")
+	fs.StringVar(&p.Storage, "storage", "", "")
+	fs.StringArrayVar(&p.Taints, "taint", []string{}, "")
+
+	return fs
+}
+
+func processKubeletArg(in string) (upcloud.KubernetesKubeletArg, error) {
+	split := strings.SplitN(in, "=", 2)
+	if len(split) < 2 {
+		return upcloud.KubernetesKubeletArg{}, fmt.Errorf("invalid kubelet-arg: %s", in)
+	}
+
+	return upcloud.KubernetesKubeletArg{
+		Key:   split[0],
+		Value: split[1],
+	}, nil
+}
+
+func processLabel(in string) (upcloud.Label, error) {
+	split := strings.SplitN(in, "=", 2)
+	if len(split) < 2 {
+		return upcloud.Label{}, fmt.Errorf("invalid label: %s", in)
+	}
+
+	return upcloud.Label{
+		Key:   split[0],
+		Value: split[1],
+	}, nil
+}
```

Comment:
Would it be good to support also empty values using something like `upctl ... --label=demo --label=env=dev` ?

Just as a alternative approach we could use `url.ParseQuery` here e.g.
```go
func processLabel(in string) (upcloud.Label, error) {
	q, err := url.ParseQuery(in)
	if err != nil {
		return upcloud.Label{}, err
	}
	for k := range q {
		return upcloud.Label{
			Key:   k,
			Value: q.Get(k),
		}, nil
	}
	return upcloud.Label{}, fmt.Errorf("unable to create label from input '%s'", in)
}
```

Reply:
Good idea! I'll edit the function and add test case for empty label value

Reply:
Added cases for `len(split)` instead of using `url.ParseQuery` as values can contain `&` characters that might be problematic when parsing query string if input is not url encoded first